src/Servers/IIS/AspNetCoreModuleV2/InProcessRequestHandler/inprocesshandler.cpp

```diff
@@ -114,6 +115,12 @@ IN_PROCESS_HANDLER::NotifyDisconnect()
     if (pManagedHttpContext != nullptr)
     {
         m_pDisconnectHandler(pManagedHttpContext);
+        // lock before notifying, this prevents the condition where m_queueNotified is already checked but
+        // the condition_variable isn't waiting yet, which would cause notify_all to NOOP and block
+        // IndicateManagedRequestComplete until a spurious wakeup
+        std::unique_lock<std::mutex> lock(m_lockQueue);
+        m_queueNotified = true;
+        m_queueCheck.notify_all();
```

Comment:
```suggestion
        m_lockQueue.unlock();
        m_queueCheck.notify_all();
```

Nit: Is it better to unlock, or is the msvc runtime smart enough to avoid waking another thread too early here?

Reply:
Looks like we should also be using a `std::lock_guard` here since we only need a `std::unique_lock` for the `std::condition_variable`